client/mapping-sync/src/sql/mod.rs

```diff
@@ -29,17 +29,20 @@ use sp_runtime::traits::{Block as BlockT, Header as HeaderT, UniqueSaturatedInto
 // Frontier
 use fp_rpc::EthereumRuntimeRPCApi;
 
-use crate::{EthereumBlockNotification, EthereumBlockNotificationSinks, SyncStrategy};
+use crate::{EthereumBlockNotification, EthereumBlockNotificationSinks, ReorgInfo, SyncStrategy};
 
 /// Defines the commands for the sync worker.
 #[derive(Debug)]
-pub enum WorkerCommand {
+pub enum WorkerCommand<Block: BlockT> {
 	/// Resume indexing from the last indexed canon block.
 	ResumeSync,
 	/// Index leaves.
 	IndexLeaves(Vec<H256>),
 	/// Index the best block known so far via import notifications.
-	IndexBestBlock(H256),
+	IndexBestBlock {
+		block_hash: H256,
+		reorg_info: Option<ReorgInfo<Block>>,
+	},
```

Comment:
Best blocks don't yet have valid information about a possible `reorg`, this should be part of `Canonicalize`.

Reply:
Addressed in https://github.com/polkadot-evm/frontier/pull/1787/commits/8139c693e699a191ee2a519d1a5e330ca1c3c626